test/e2e/nutanix_client.go

```diff
@@ -0,0 +1,70 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"flag"
+	nutanixClientHelper "github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pkg/client"
+	prismGoClient "github.com/nutanix-cloud-native/prism-go-client/pkg/nutanix"
+	prismGoClientV3 "github.com/nutanix-cloud-native/prism-go-client/pkg/nutanix/v3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"os"
+	"strconv"
+)
+
+var (
+	nutanixUser     = os.Getenv("NUTANIX_USER")
+	nutanixPassword = os.Getenv("NUTANIX_PASSWORD")
+	nutanixEndpoint string
+	nutanixPort     string
+	nutanixInsecure string
+	creds           prismGoClient.Credentials
+	nutanixClient   *prismGoClientV3.Client
+	err             error
+)
+
+func init() {
+	flag.StringVar(&nutanixEndpoint, "e2e.nutanixEndpoint", os.Getenv("NUTANIX_ENDPOINT"), "the Nutanix Prism Central used for e2e tests")
+	flag.StringVar(&nutanixPort, "e2e.nutanixPort", os.Getenv("NUTANIX_PORT"), "the Nutanix Prism Central port used for e2e tests")
+	flag.StringVar(&nutanixInsecure, "e2e.nutanixInsecure", os.Getenv("NUTANIX_INSECURE"), "Ignore certificate checks for e2e tests")
```

Comment:
consider using https://github.com/spf13/viper and https://github.com/spf13/cobra to avoid using globals.